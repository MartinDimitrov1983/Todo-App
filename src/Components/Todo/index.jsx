import React from 'react';
import PropTypes from 'prop-types';
import { CheckBox, Button } from '..';
import styles from './index.module.css';
import { ADD_COLOR, DELETE, RESOLVE, WHITE_COLOR, DONE } from '../../Constants';

const Todo = ({
  text,
  onDelete,
  isResolved,
  onResolveSingleTodo,
  color,
  onAddColor,
}) => {
  return (
    <div
      className={styles.container}
      style={{ background: color ? color : WHITE_COLOR }}
    >
      <div className={styles.text}>{text}</div>
      <div className={styles.checkbox}>
        <CheckBox
          label={RESOLVE}
          checked={isResolved}
          onChange={onResolveSingleTodo}
        />
        {isResolved && <div className={styles.done}>{DONE}</div>}
      </div>
      <div className={styles.buttons}>
        <Button text={ADD_COLOR} onClick={onAddColor} />
        <Button text={DELETE} onClick={() => onDelete()} />
      </div>
    </div>
  );
};

Todo.propTypes = {
  text: PropTypes.string,
  onDelete: PropTypes.func,
  isResolved: PropTypes.bool,
  resolveSingleTodo: PropTypes.func,
  color: PropTypes.string,
  addColor: PropTypes.func,
};

export default Todo;
