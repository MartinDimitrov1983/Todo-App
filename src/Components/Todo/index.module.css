.container {
  display: flex;
  align-items: center;
  margin-bottom: 25px;
  border-radius: 5px;
  background-color: #fff;
  padding: 15px 20px;
  filter: drop-shadow(0 0 0.2rem rgb(193, 193, 193));
}

.text {
  flex: 2;
  text-transform: capitalize;
}

.checkbox {
  display: flex;
  flex: 2;
  gap: 20px;
}

.done {
  display: flex;
  align-items: center;
}

.buttons {
  display: flex;
  gap: 20px;
}
