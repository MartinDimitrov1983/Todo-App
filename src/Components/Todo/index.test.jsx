import React from 'react';
import { render } from '@testing-library/react';

import Todo from './index';

test('Snapshot Todo', () => {
  const { asFragment } = render(
    <Todo
      text="Test"
      onDelete={() => {}}
      isResolved={true}
      onResolveSingleTodo={() => {}}
      color="Test"
      onAddColor={() => {}}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
