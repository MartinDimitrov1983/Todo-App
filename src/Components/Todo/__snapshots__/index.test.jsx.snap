// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Todo 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="text"
    >
      Test
    </div>
    <div
      class="checkbox"
    >
      <div
        class="checkBoxWrapper"
      >
        <label
          class="label"
        >
          <input
            checked=""
            class="input"
            type="checkbox"
          />
          <span>
            Resolve
          </span>
        </label>
      </div>
      <div
        class="done"
      >
        Done
      </div>
    </div>
    <div
      class="buttons"
    >
      <button
        class="button"
        type="button"
      >
        <div>
          Add Color
        </div>
      </button>
      <button
        class="button"
        type="button"
      >
        <div>
          Delete
        </div>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
